packages/forms/src/UIForm/customFormats.js

```diff
@@ -1,18 +1,23 @@
 const emailRegExp = /^[a-zA-Z][a-zA-Z0-9-.]+@[a-zA-Z-]+\.[a-zA-Z-]+$/;
 const urlHttpOrHttpsRegExp = /^(http|https):\/\/[a-zA-Z0-9.\-/_]+$/;
 /* eslint-disable max-len */
+// Format usable in regex101 : (?<http>^http\:\/\/)(?<hostHTTP>[a-zA-Z0-9\.\-_]+)(?<portHTTP>:[0-9]+)?(?<pathHTTP>\/[a-zA-Z0-9\/\.\-_]+)(?<extensionHTTP>\.git)?(?<slashHTTP>\/)?$
+const urlGitProtocoltHttp = /(^http:\/\/)([a-zA-Z0-9.\-_]+)(:[0-9]+)?(\/[a-zA-Z0-9/.\-_]+)(\.git)?(\/)?$/;
 // Format usable in regex101 : (?<http>^http(s)?\:\/\/)(?<hostHTTP>[a-zA-Z0-9\.\-_]+)(?<portHTTP>:[0-9]+)?(?<pathHTTP>\/[a-zA-Z0-9\/\.\-_]+)(?<extensionHTTP>\.git)?(?<slashHTTP>\/)?$
-const urlGitProtocoltHttp = /(^http(s)?:\/\/)([a-zA-Z0-9.\-_]+)(:[0-9]+)?(\/[a-zA-Z0-9/.\-_]+)(\.git)?(\/)?$/;
+const urlGitProtocoltHttps = /(^https?:\/\/)([a-zA-Z0-9.\-_]+@)?([a-zA-Z0-9.\-_]+)(:[0-9]+)?(\/[a-zA-Z0-9/.\-_]+)(\.git)?(\/)?$/;
 // Format usable in regex101 : (?<ssh>^ssh\:\/\/)(?<userSSH>[a-zA-Z0-9\.\-_]+\@)?(?<hostSSH>[a-zA-Z0-9\.\-_]+)(?<portSSH>:[0-9]+)?(?<pathSSH>\/[a-zA-Z0-9\/\.\-_~]+)(?<extensionSSH>\.git)(?<slashSSH>\/)?$
 const urlGitProtocolSsh = /(^ssh:\/\/)([a-zA-Z0-9.\-_]+@)?([a-zA-Z0-9.\-_]+)(:[0-9]+)?(\/[a-zA-Z0-9/.\-_~]+)(\.git)(\/)?$/;
+// Format usable in regex101 : (?<userSSH>^[a-zA-Z0-9\.\-_]+\@)(?<hostSSH>[a-zA-Z0-9\.\-_]+)(:)(?<portSSH>[0-9]+)?(?<pathSSH>[a-zA-Z0-9\.\-_~]+)(?<pathSSH2>\/[a-zA-Z0-9\/\.\-_~]+)(?<extensionSSH>\.git)(?<slashSSH>\/)?$
+const urlGitProtocolSsh2 = /(^[a-zA-Z0-9.\-_]+@)([a-zA-Z0-9.\-_]+)(:)([0-9]+)?([a-zA-Z0-9.\-_~]+)(\/[a-zA-Z0-9/.\-_~]+)(\.git)(\/)?$/;
```

Comment:
Isn't it possible to update `urlGitProtocolSsh` to include that case?

Reply:
not because with the second case we need to start the path with ":" if there is no port

Reply:
Also the idea is to not create not maintainable regexp with one who can manage all cases